crates/bevy_ui/src/flex/convert.rs

```
@@ -33,30 +37,34 @@ pub fn from_val_size(
-        size: from_val_size(scale_factor, value.size),
-        min_size: from_val_size(scale_factor, value.min_size),
-        max_size: from_val_size(scale_factor, value.max_size),
-        aspect_ratio: match value.aspect_ratio {
-            Some(value) => taffy::number::Number::Defined(value),
-            None => taffy::number::Number::Undefined,
-        },
+pub fn from_flex_layout(scale_factor: f64, value: FlexLayoutQueryItem<'_>) -> taffy::style::Style {
+    if let Layout::FlexLayout(flex_layout) = value.layout {
+        taffy::style::Style {
+            display: taffy::style::Display::Flex,
+            position_type: (*value.position_type).into(),
+            flex_direction: flex_layout.flex_direction.into(),
+            flex_wrap: flex_layout.wrap.into(),
+            align_items: flex_layout.align_items.into(),
+            align_self: flex_layout.align_self.into(),
+            align_content: flex_layout.align_content.into(),
+            justify_content: flex_layout.justify_content.into(),
+            position: from_rect(scale_factor, value.offset.0),
+            margin: from_rect(scale_factor, value.spacing.margin),
+            padding: from_rect(scale_factor, value.spacing.padding),
+            border: from_rect(scale_factor, value.spacing.border),
+            flex_grow: flex_layout.grow,
+            flex_shrink: flex_layout.shrink,
+            flex_basis: from_val(scale_factor, flex_layout.basis),
+            size: from_val_size(scale_factor, value.size_constraints.suggested),
+            min_size: from_val_size(scale_factor, value.size_constraints.min),
+            max_size: from_val_size(scale_factor, value.size_constraints.max),
+            aspect_ratio: match value.size_constraints.aspect_ratio {
+                Some(value) => taffy::number::Number::Defined(value),
+                None => taffy::number::Number::Undefined,
+            },
+        }
+    } else {
+        panic!("Not quite sure what to do here?");
```

Any good ideas on what to do in the event that we call this with something that is not `Layout::FlexLayout`?

Pretty sure panic is not the right approach but not sure about alternatives.

---

If it's an invariant that must be fulfilled then a panic is the right approach IMO. Better to fail early.

As far as I see it's also not user-facing, so we don't have to account for user misuse here.

---

The panic message should at least be changed so that it makes some sense if it ever happens.

---

This is reachable if someone changes the `Layout` component value for a node that's otherwise fully Flexbox-compliant. I would go for a `panic!("")` that specifies that a Flexbox-compliant UI node with a non-flexbox `Layout` value was found and that it's not supported.